src/Helper/Product/Utility.php

```diff
@@ -140,7 +140,7 @@ public function isVisibleInCategories($product): bool
             Visibility::VISIBILITY_IN_CATALOG
         ];
 
-        return in_array($product->getVisibility(), $validVisibilities, true);
+        return in_array($product->getVisibility(), $validVisibilities, false);
```

Comment:
According to our coding standards, I think we should cast the type and then have the type sensitive comparison.
At least we did it this way in the past

Reply:
kept the strict comparison